avocado/utils/network/interfaces.py

```diff
@@ -71,10 +73,17 @@ def _get_interface_details(self, version=4):
                     return item
             raise NWException("Interface not found")
         except (NWException, json.JSONDecodeError):
-            msg = "Unable to get IP address on interface {}".format(self.name)
+            msg = "Unable to get the details of interface {}".format(self.name)
             log.error(msg)
             raise NWException(msg)
 
+    def _get_bondinterface_details(self, version=4):
+        cmd = "cat /sys/class/net/{}/bonding/mode /sys/class/net/{}/bonding/slaves".format(self.name, self.name)
+        try:
+            return run_command(cmd, self.host)
```

Comment:
Here you are returning the raw output, leaving to upper layer methods to handle this output. Would be nice to return here the specific information or a dict so caller methods wouldn't need to do splitlines(), split() and stuff like that.